wee_slack.py

```diff
@@ -3404,10 +3432,28 @@ def command_upload(data, current_buffer, args):
     if ' ' in file_path:
         file_path = file_path.replace(' ', '\ ')
 
-    command = 'curl -F file=@{} -F channels={} -F token={} {}'.format(file_path, channel.identifier, team.token, url)
+    # only http proxy is currenlty supported
+    proxy = ""
+    proxy_name = proxy = w.config_string(weechat.config_get('weechat.network.proxy_curl'))
+    if not proxy_name:
+        proxy_string = "weechat.proxy.{}".format(proxy_name)
+        proxy_type = w.config_integer(weechat.config_get("{}.type".format(proxy_string)))
+        if proxy_type == "http":
+            proxy_address = w.config_string(weechat.config_get("{}.address".format(proxy_string)))
+            proxy_port = w.config_string(weechat.config_get("{}.port".format(proxy_string)))
+            proxy_user = w.config_string(weechat.config_get("{}.username".format(proxy_string)))
+            proxy_password = w.config_string(weechat.config_get("{}.password".format(proxy_string)))
+            proxy = "--proxy "
```

Comment:
These functions are called a few times, can you factor it into a function instead that just returns the settings?